### PR TITLE
[TR] Migration sf 2.7

### PR DESCRIPTION
**Pull request by skeleton**

### CODE REVIEW

File: app/SymfonyRequirements.php [hunk 1]
```diff
@@ -41,25 +41,25 @@ class Requirement
     /**
      * Constructor that initializes the requirement.
```
Comment (jjanvier) [inline app/SymfonyRequirements.php]: this change needs to be check by @BitOne to be sure everything's OK from the ops POV

File: composer.json [hunk 2]
```diff
@@ -18,9 +18,15 @@
...
         "akeneo/batch-bundle": "0.4.*",
         "akeneo/measure-bundle": "0.2.*",
-        "apy/jsfv-bundle": "2.0.1",
-        "ass/xmlsecurity": "1.0.0",
-        "doctrine/annotations": "1.2.3",
+        "symfony/assetic-bundle": "2.3.1",
+        "symfony/icu": "1.1.0",
+        "symfony/monolog-bundle": "~2.4",
+        "symfony/swiftmailer-bundle": "2.3.8",
+        "symfony/symfony": "2.7.*",
```
Comment (jjanvier) [inline composer.json]: set it to a real tag

File: composer.json [hunk 3]
```diff
@@ -18,9 +18,15 @@
         "php": ">=5.4.4",
         "akeneo/batch-bundle": "0.4.*",
         "akeneo/measure-bundle": "0.2.*",
-        "apy/jsfv-bundle": "2.0.1",
-        "ass/xmlsecurity": "1.0.0",
-        "doctrine/annotations": "1.2.3",
+        "symfony/assetic-bundle": "2.3.1",
```
Comment (jjanvier) [inline composer.json]: order everything from a to z please

File: composer.json [hunk 4]
```diff
@@ -47,25 +55,15 @@
...
-        "sensio/generator-bundle": "2.3.5",
-        "stof/doctrine-extensions-bundle": "1.1.0",
-        "symfony/assetic-bundle": "2.3.1",
-        "symfony/icu": "1.1.0",
-        "symfony/monolog-bundle": "2.3.0",
-        "symfony/swiftmailer-bundle": "2.3.8",
-        "symfony/symfony": "2.3.23",
-        "twig/extensions": "1.0.1",
-        "twig/twig": "<1.18.1"
+        "oro/platform": "dev-bump-symfony-marie"
```
Comment (jjanvier) [inline composer.json]: don't forget it :)

File: src/Pim/Bundle/CatalogBundle/DependencyInjection/PimCatalogExtension.php [hunk 5]
```diff
@@ -43,41 +42,10 @@ public function load(array $configs, ContainerBuilder $container)
...
-        $this->loadValidationFiles($container);
         $this->loadStorageDriver($container);
     }
 
     /**
-     * Loads the validation files from all bundles
-     *
-     * @param ContainerBuilder $container
-     */
-    protected function loadValidationFiles(ContainerBuilder $container)
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/DependencyInjection/PimCatalogExtension.php]: done by default now ? so good :+1:

File: src/Pim/Bundle/CatalogBundle/PimCatalogBundle.php [hunk 6]
```diff
@@ -5,6 +5,7 @@
 use Akeneo\Bundle\StorageUtilsBundle\AkeneoStorageUtilsBundle;
 use Akeneo\Bundle\StorageUtilsBundle\DependencyInjection\Compiler\ResolveDoctrineTargetRepositoryPass;
 use Oro\Bundle\EntityBundle\DependencyInjection\Compiler\DoctrineOrmMappingsPass;
+use Pim\Bundle\CatalogBundle\DependencyInjection\Compiler\LoadValidationFilesPass;
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/PimCatalogBundle.php]: why ?

File: src/Pim/Bundle/CatalogBundle/Resources/config/validators.yml [hunk 7]
```diff
@@ -226,13 +228,19 @@ services:
...
             - [ 'addMetadataFactory', [ '@validator.mapping.class_metadata_factory' ] ]
 
+    pim_catalog.validator.context.factory:
+        public: false
+        class:  Symfony\Component\Validator\Context\ExecutionContextFactory
+        arguments:
+            - '@translator.default'
+            - %validator.translation_domain%
+
     pim_catalog.validator.product:
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Resources/config/validators.yml]: bc here

File: composer.json [hunk 8]
```diff
@@ -47,25 +55,15 @@
...
         "liip/imagine-bundle": "0.17.1",
         "monolog/monolog": "1.10.0",
-        "nelmio/api-doc-bundle": "2.4.4",
+        "nelmio/api-doc-bundle": "2.9.0",
         "oneup/flysystem-bundle": "1.1.0",
-        "oro/platform": "dev-pim-1.4@dev",
-        "sensio/distribution-bundle": "2.3.7",
-        "sensio/framework-extra-bundle": "2.3.4",
-        "sensio/generator-bundle": "2.3.5",
-        "stof/doctrine-extensions-bundle": "1.1.0",
```
Comment (jjanvier) [inline composer.json]: for the record, why do we remove stof/doctrine-extensions-bundle

what was it use for ?

**Comment by jjanvier** [comment]
GTM once comments fixed
check with @nidup and @BitOne before merging if it's still ok

**Comment by jjanvier** [comment]
missing changelog and detailed upgrade.md

File: composer.json (hunk 8 above)
Comment (skeleton) [inline composer.json]: stof/doctrine-extensions-bundle fixes gedmo to v2.3.1 which contains deprecated methods/classes.

As, we use only stof bundle to configure gedmo, I removed it, upgraded gedmo to v2.4.3 and configured it in gedmo_doctrine_extensions.yml

File: src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/EditCommonAttributesType.php [hunk 9]
```diff
@@ -95,6 +96,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'multiple' => true,
                     'expanded' => false,
                     'group_by' => 'group.label',
+                    'choice_value' => function (AttributeInterface $attribute) {
+                        // Cast id to string to be compatible with ChoiceView
+                        return (string) $attribute->getId();
```
Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/EditCommonAttributesType.php]: ok

**Comment by nidup** [comment]
GTM

**Pull request merged**